### PR TITLE
Replace send otp by email by send URL by

### PR DESCRIPTION
**Pull request by Mohadgb**
- Refactored AuthController to improve security and maintainability:

- Replaced OTP-based password reset with secure one-time reset link.

- Added robust error handling and Sentry integration for all auth operations.

- Implemented dummy hash comparison in authenticate to prevent timing attacks.

- Enforced strong password validation during registration and password reset.

- Updated all tests to match the new implementation:

- Converted OTP tests to reset link/token tests.
 
- Ensured proper coverage for authenticate, registerUser, forgotPassword, resetPassword, and changePassword.

- Included edge cases and error handling tests for database failures.

### CODE REVIEW

File: Back-End/controllers/mondoDBControllers/AuthController.ts [hunk 1]
```diff
@@ -58,141 +58,107 @@ export class AuthController {
...
       const user = await User.findOne({ email }).exec();
+      if (!user) return { message: 'If the email exists, a reset link has been sent.' };
 
-      // Generate OTP - 4 digits
-      const otp = randomInt(1000, 10000).toString();
-      const otpExpire = new Date(
-        Date.now() + this.OTP_EXPIRY_MINUTES * 60 * 1000,
-      );
+      const resetToken = randomBytes(32).toString('hex');
+      const resetLink = `${process.env.FRONTEND_URL}/reset-password/${resetToken}`;
```
Comment (iKozay) [inline Back-End/controllers/mondoDBControllers/AuthController.ts]: make sure FRONTEND_URL is defined

File: Back-End/controllers/mondoDBControllers/AuthController.ts [hunk 2]
```diff
@@ -58,141 +58,107 @@ export class AuthController {
...
+      const expire = new Date(Date.now() + this.RESET_EXPIRY_MINUTES * 60 * 1000);
 
-      if (user) {
-        // Save OTP and expiry to user record
-        user.otp = otp;
-        user.otpExpire = otpExpire;
-        await user.save();
+      user.resetToken = resetToken;
```
Comment (iKozay) [inline Back-End/controllers/mondoDBControllers/AuthController.ts]: Instead of saving it to the db, use Redis

File: Back-End/controllers/mondoDBControllers/AuthController.ts [hunk 3]
```diff
@@ -58,141 +58,107 @@ export class AuthController {
...
         return false;
       }
 
-      // Set new password (already hashed from frontend) and clear OTP
-      user.password = newPassword; // password is already hashed from frontend
-      user.otp = '';
-      user.otpExpire = new Date(0);
+      user.password = newPassword; // already hashed from frontend
+      user.resetToken = '';
+      user.resetTokenExpire = new Date(0);
```
Comment (iKozay) [inline Back-End/controllers/mondoDBControllers/AuthController.ts]: you can set it to 0 instead of doing new Date

File: Back-End/models/User.ts [hunk 4]
```diff
@@ -12,10 +12,11 @@ const UserSchema = new Schema({
   fullname: { type: String, required: true },
   degree: { type: String, ref: 'Degree' },
   type: { type: String, enum: ['student', 'advisor', 'admin'], required: true },
-  otp: { type: String, default: null },
-  otpExpire: { type: Date, default: null },
+  resetToken: { type: String, default: null },
```
Comment (iKozay) [inline Back-End/models/User.ts]: as mentionned in comment above, i recommend to use redis to save this value

**Review by iKozay — CHANGES_REQUESTED** [review]
Good work. Just a few changes to do + use Redis to store the token

File: Back-End/models/User.ts [hunk 5]
```diff
@@ -13,8 +13,9 @@ const UserSchema = new Schema({
...
```
Comment (iKozay) [inline Back-End/models/User.ts]: If you are saving the token in redis, you can safely remove those fields

File: Back-End/models/User.ts [hunk 6]
```diff
@@ -13,8 +13,8 @@ const UserSchema = new Schema({
   fullname: { type: String, required: true },
   degree: { type: String, ref: 'Degree' },
   type: { type: String, enum: ['student', 'advisor', 'admin'], required: true },
-  otp: { type: String, default: null },
-  otpExpire: { type: Date, default: null },
+  resetTokenExpire: { type: Date, default: null },
```
Comment (iKozay) [inline Back-End/models/User.ts]: there is no need for `resetTokenExpire`. It can be removed
